internal/controller/policyserver_controller_deployment.go

```diff
@@ -220,28 +225,111 @@ func (r *PolicyServerReconciler) updatePolicyServerDeployment(policyServer *poli
+				Name: otelClientCertificateVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: r.OtelClientCertificateSecret,
+					},
+				},
+			})
+			admissionContainer.VolumeMounts = append(admissionContainer.VolumeMounts, corev1.VolumeMount{
+				Name:      otelClientCertificateVolumeName,
+				ReadOnly:  true,
+				MountPath: otelClientCertificateVolumePath,
+			})
+
+			admissionContainer.Args = append(admissionContainer.Args, "--otlp-client-certificate="+otelClientCertificateVolumePath+"/tls.crt")
+			admissionContainer.Args = append(admissionContainer.Args, "--otlp-client-certificate-key="+otelClientCertificateVolumePath+"/tls.key")
+		}
+		// As the controller is sending data to remote otel collector, we need
+		// to replicate the env vars to the policy server deployment. Thus, it
+		// will be able to send data to the same collector.
+		otelEnvVarToReplicate := []string{
+			"OTEL_EXPORTER_OTLP_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_CLIENT_KEY",
+			"OTEL_EXPORTER_OTLP_COMPRESSION",
+			"OTEL_EXPORTER_OTLP_HEADERS",
+			"OTEL_EXPORTER_OTLP_INSECURE",
+			"OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_METRICS_CLIENT_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_METRICS_CLIENT_KEY",
+			"OTEL_EXPORTER_OTLP_METRICS_COMPRESSION",
+			"OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION",
+			"OTEL_EXPORTER_OTLP_METRICS_ENDPOINT",
+			"OTEL_EXPORTER_OTLP_METRICS_HEADERS",
+			"OTEL_EXPORTER_OTLP_METRICS_INSECURE",
+			"OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE",
+			"OTEL_EXPORTER_OTLP_METRICS_TIMEOUT",
+			"OTEL_EXPORTER_OTLP_TIMEOUT",
+			"OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_TRACES_CLIENT_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_TRACES_CLIENT_KEY",
+			"OTEL_EXPORTER_OTLP_TRACES_COMPRESSION",
+			"OTEL_EXPORTER_OTLP_TRACES_ENDPOINT",
+			"OTEL_EXPORTER_OTLP_TRACES_HEADERS",
+			"OTEL_EXPORTER_OTLP_TRACES_INSECURE",
+			"OTEL_EXPORTER_OTLP_TRACES_TIMEOUT",
+		}
+		// "OTEL_EXPORTER_OTLP_ENDPOINT" is not replicated because the endpoint is passed using CLI flag.
+		for _, envVar := range otelEnvVarToReplicate {
+			if value := os.Getenv(envVar); value != "" {
+				envvar := corev1.EnvVar{Name: envVar, Value: value}
+				if index := envVarsContainVariable(admissionContainer.Env, envVar); index >= 0 {
+					admissionContainer.Env[index] = envvar
+				} else {
+					admissionContainer.Env = append(admissionContainer.Env, envvar)
+				}
+			}
+		}
+		// The rust library needs to have the scheme while the go one fails.
+		// TODO- Try to use WithEndpointURL here in the controller. Maybe we can remove the
+		// following string manipulation.
+		policyServerEndpoint := "http://" + r.OtelCollectorEndpoint
+		admissionContainer.Args = append(admissionContainer.Args, "--otlp-endpoint="+policyServerEndpoint)
```

Comment:
This is fragile. Let's coordinate with @fabriziosestito to handle it better. Having the `http` hard coded is also bad (no worries, I understand this is a draft); we have to take into account the `https` scenario